src/shared/reducers/message.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2015-present, CWB SAS
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import createReducer from "./createReducer";
+import {
+  API_ADMIN,
+  API_SETADMINPARAMETERS,
+  API_USERPROFILE,
+  AUTH_SIGNIN,
+  AUTH_SIGNOUT,
+  APP_SETMESSAGE,
+  APP_REMOVEMESSAGE,
+  FETCH_FAILURE,
+} from "../actions/constants";
+
+export const initialState = {
+  message: null,
+};
+
+export const handlers = {
+  /* App section */
+  [API_ADMIN + FETCH_FAILURE]: (state, { error }) => ({
+    ...state,
+    messages: error,
+  }),
+
+  [API_SETADMINPARAMETERS + FETCH_FAILURE]: (state, { error }) => ({
+    ...state,
+    messages: error,
+  }),
+
+  /* Auth section */
+  [AUTH_SIGNIN + FETCH_FAILURE]: (state, { error }) => ({
+    ...state,
+    messages: error,
+  }),
+  [AUTH_SIGNOUT + FETCH_FAILURE]: (state, { error }) => ({
+    ...state,
+    messages: error,
+  }),
+
+  /* User section */
+  [API_USERPROFILE + FETCH_FAILURE]: (state, { error }) => ({
+    ...state,
+    messages: error,
+  }),
+
+  [APP_SETMESSAGE]: (state, { message }) => ({
+    ...state,
+    message,
+  }),
+
+  [APP_REMOVEMESSAGE]: state => ({
+    ...state,
+    message: null,
+  }),
+};
```

Comment:
To avoid duplicate code, you should switch to a `switch`/`case` based reducer:

```
function reducer(state = initialState, action) {
  switch (action.type) {
  }
}
```

Reply:
we agreed on not doing this because we need to extend this reducer in opa/front